### PR TITLE
[FEATURE] Refactor project related routes, add authorization, juries and grades.

### LINKED ISSUE #4: [FEATURE] Add authorization on routers.
**Is your feature request related to a problem? Please describe.**
Users should be authorized to access only endpoints accessible to their roles.

**Describe the solution you'd like**
Add authorization on routers to ensure that normal users cannot access PM or professor endpoints, using JWT for access and refresh tokens.
Create a middleware to be used by all routers requiring valdation.

**Describe alternatives you've considered**
N/A

**Provide usage examples**
All requests needing authorization will define the `Authorization` header, and will set its value to `Bearer <token>`, where the token is the access token received by logging in.

**Additional context**
N/A

### LINKED ISSUE #7: [FEATURE] Create project management infrastructure.
**Is your feature request related to a problem? Please describe.**
Users should be able to create, update and delete projects. Professors should be able to add and remove team members to a project.

**Describe the solution you'd like**

**Describe alternatives you've considered**
N/A

**Provide usage examples**
N/A

**Additional context**
N/A

### LINKED ISSUE #11: [FEATURE] Add authorization for grading.
**Is your feature request related to a problem? Please describe.**
Only jury members should be able to add or modify grades and only their own grades.

**Describe the solution you'd like**
Validate that the user making the request is a jury member to the project they want to modify and that they can only modify their grade.

**Describe alternatives you've considered**
N/A

**Provide usage examples**
N/A

**Additional context**
N/A

### LINKED ISSUE #12: [FEATURE] Prevent grade modification beyond a predefined date.
**Is your feature request related to a problem? Please describe.**
Grades should only be able to be modified for only a certain amount of time.

**Describe the solution you'd like**
Ensure that users can modify their grades only for a certain amount of time, based on the creation date of the grade.

**Describe alternatives you've considered**
N/A

**Provide usage examples**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by b0tw**
Resolves #4, resolves #7, resolves #11 , resolves #12.

Refactored project related routes

### CODE REVIEW

File: server/routers/teams.js [hunk 1]
```diff
@@ -5,137 +5,122 @@ const Team = context.Team
...
-        return res.status(400).send(apiError.InvalidRequest);
-    }
+  if (isNaN(id)) {
+    return res.status(400).send(apiError.InvalidRequest);
+  }
+
+  let team = await context.Team.findOne({ where: { id: id }, include: [{ model: context.User, attributes: ['id', 'surname', 'name'] }] });
+  if (team == null) {
+    return res.status(400).send(apiError.InvalidRequest);
+  }
```
Comment (filipdutescu) [inline server/routers/teams.js]: really good you implemented this or I would've requested it :laughing:

File: server/routers/teams.js [hunk 2]
```diff
@@ -5,137 +5,122 @@ const Team = context.Team
...
+  const name = req.body.name
+  const project_name = req.body.project_name
+
+  //get the username from the session
+  const username = req.username
+
+  //search in the db the user
+  const user = await User.findOne({ where: username })
+  //if the user is a professor, the addition can happen
+  if (user.is_professor === true) {
```
Comment (filipdutescu) [inline server/routers/teams.js]:
```suggestion
  if (user && user.is_professor === true) {
```

File: server/routers/teams.js [hunk 3]
```diff
@@ -5,137 +5,122 @@ const Team = context.Team
...
-            return res.status(400).send({ "message": "Something happened at deletion" })
-        }
+  //get the username from the session
+  const username = req.username
+
+  //search in the db the user
+  const user = await User.findOne({ where: username })
+
+  //if the user is a professor, the deletion can happen
+  if (user.is_professor === true) {
```
Comment (filipdutescu) [inline server/routers/teams.js]:
```suggestion
  if (user && user.is_professor === true) {
```

File: server/routers/teams.js [hunk 4]
```diff
@@ -5,137 +5,122 @@ const Team = context.Team
...
+      const team = await Team.findOne({ where: { id }, include: [context.User, context.Project] })
+      team.removeProjects(team.Projects);
+      team.removeUsers(team.Users);
+      await team.destroy()
+      return res.status(200).send({ "message": "Team was deleted" })
     }
-    else {
-        return res.status(400).send({ "message": apiError.Unauthorized })
+    catch (err) {
+      return res.status(400).send({ "message": "Something happened at deletion" })
```
Comment (filipdutescu) [inline server/routers/teams.js]: All such responses should be:

```suggestion
      return res.status(500).json(apiError.InternalError);
```

**Review by filipdutescu — CHANGES_REQUESTED** [review]

**Comment by filipdutescu** [comment]
I'll handle the remaining issues in this PR

**Comment by b0tw** [comment]
Looks good

**Review by filipdutescu — APPROVED** [review]

**Pull request merged**